### PR TITLE
show "add ousd coin to wallet" modal

### PR DESCRIPTION
**Pull request by sparrowDom**
After the user mints OUSD for the first time, and the OUSD balance finishes animating. We show the modal that prompts users to add OUSD coin to their wallet. 

If the provider is Metamask, we just show a button that automatically pre-fills all the needed data and user only needs to approve Metamask prompt.
If the provider is any other we show the contract address + decimals information. 

**related issue:** https://github.com/OriginProtocol/origin-dollar/issues/270

### CODE REVIEW

**Comment by micahalcorn** [comment]
Most of the mobile wallets that I've tested detect the token automatically. So it's unnecessary (and potentially confusing) to show this information in those cases. I was proposing that we only show this for MetaMask users. I suppose the more thorough approach would be to manually blacklist all the environments that we know definitively detect the token.

File: dapp/src/utils/web3.js [hunk 1]
```diff
@@ -44,7 +52,36 @@ export function truncateAddress(address) {
...
+    },
+    console.log
+  )
+}
+
+/* status of token wallets and OUSD:
+ * https://docs.google.com/spreadsheets/d/1bunkxBxfkAVz9C14vAFH8CZ53rImDNHTXp94AOEjpq0/edit#gid=1608902436
+ */
+export function providersNotAutoDetectingOUSD() {
+  return ['metamask', 'trust', 'alphawallet', 'mist', 'parity']
```
Comment (sparrowDom) [inline dapp/src/utils/web3.js:77]: @micahalcorn added this bit of functionality where we whitelist the providers where we show the "add OUSD modal". We can cross them out from here when the wallets pick up OUSD, or add new ones if we see in analytics that some provider is popular and isn't auto detecting OUSD.

**Comment by sparrowDom** [comment]
@shahthepro merging this in. Happy to do any changes later in case need be.

**Review by shahthepro — COMMENTED** [review]
Sorry just saw this PR. Looks good to me. LGTM

**Pull request merged**